sdk/eidolon_ai_sdk/system/agent_controller.py

```diff
@@ -251,13 +251,17 @@ async def agent_event_stream(self, handler, process, last_state, **kwargs) -> As
             logger.info(f"Process {process.record_id} was cancelled")
             if not ended:
                 if not transitioned:
+                    # update the process object as it might have been updated by the function
+                    process = await ProcessDoc.find_one(query={"_id": process.record_id})
```

Comment:
Can we remove this and instead update just the state/updated time here? The point was to avoid race conditions, which is important on the initial state transition, but less valuable on the back side. And here it doesn't matter at all since we are fetching/updating.

Reply:
this force flag / method would be used for every comment

# update the process object as it might have been updated by the function